specifyweb/workbench/upload/upload_table.py

```diff
@@ -55,6 +56,28 @@ def to_json(self) -> Dict:
 
     def unparse(self) -> Dict:
         return { 'baseTableName': self.name, 'uploadable': self.to_json() }
+    
+class DeferredScopeUploadTable(NamedTuple):
+    name: str
+    overrideScope: Optional[Dict[str, Optional[int]]]
+    wbcols: Dict[str, ColumnOptions]
+    static: Dict[str, Any]
+    toOne: Dict[str, Uploadable]
+    toMany: Dict[str, List[ToManyRecord]]
+
+    related_key: str
+    relationship_name: str
+    filter_field: str
+
+    def apply_scoping(self, collection, deffer: bool = True) -> "ScopedUploadTable":
+        if not deffer:
+            from .scoping import apply_scoping_to_uploadtable as apply_scoping
```

Comment:
once inherited from UploadTable, could just call `super().apply_scoping(...)` in there

Reply:
This is more tricky than it seems. 
See my breakdown in https://github.com/specify/specify7/pull/3240#discussion_r1149507690